<h2>You can now run Alphafold from your browser in a more &nbsp;interactive way using Seqera platform and nextflow.</h2><h3>Details <a href="/sites/rc2/SitePages/Protein-Structure-Prediction-using-Seqera-Platform.aspx" data-interception="off" target="_blank" rel="noopener noreferrer">here</a>.</h3>
<h2>AlphaFold</h2><p><i><strong>For Licence please check&nbsp;</strong></i><a href="https://github.com/deepmind/alphafold#license-and-disclaimer" data-interception="off" target="_blank" rel="noopener noreferrer"><i><strong>License and disclaimer</strong></i></a></p><p>Any publication that discloses findings arising from using the source code or the model parameters should&nbsp;<a href="https://github.com/deepmind/alphafold#citing-this-work">cite</a>&nbsp;the&nbsp;<a href="https://doi.org/10.1038/s41586-021-03819-2">AlphaFold paper</a>&nbsp;and, if applicable, the&nbsp;<a href="https://www.biorxiv.org/content/10.1101/2021.10.04.463034v1">AlphaFold-Multimer paper</a>.</p><p>To be kept updated with news about AlphaFold on Milton</p>
<h4><span class="fontColorBlue">Contents on this page:</span></h4><ul><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#alphafold-v2.3">​​​AlphaFold v2.3</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#alphafold-v2.2.0.0">AlphaFold v2.2</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#alphafold-v2.1">AlphaFold v2.1</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#running-alphafold2.1-on-milton-tutorial">Running AlphaFold 2.1 on Milton tutorial</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#alphafold-v2.0">AlphaFold v2.0</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#running-alphafold2.0-on-milton-tutorial">Running AlphaFold 2.0 on Milton tutorial</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#output-directory-structure">Output directory structure</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#amber-relaxation-and-homo-oligomers">Amber relaxation and homo-oligomers</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#alphafold-command">AlphaFold command</a></li><li><a href="/sites/rc2/SitePages/AlphaFold.aspx#discussion-group-meeting-on">Discussion group meeting</a>​​​​​​​</li></ul>
<h2><span class="fontColorThemeSecondary">AlphaFold v2.3</span></h2><p><a href="https://github.com/deepmind/alphafold/releases/tag/v2.3.1" data-interception="off" target="_blank" rel="noopener noreferrer"><strong>Alphafold v2.3.1</strong></a> and <a href="https://github.com/deepmind/alphafold/releases/tag/v2.3.2" data-interception="off" target="_blank" rel="noopener noreferrer"><strong>Alphafold v2.3.2</strong></a><strong>&nbsp;</strong>are now available on Milton.</p><pre>module av alphafold\n------------ /stornext/System/data/modulefiles/structbio/its -----------------------------------\nalphafold/2.0.0 alphafold/2.0.1 alphafold/2.0.2 alphafold/2.1.1.0 alphafold/2.1.2.0 alphafold/2.2.0.0 alphafold/2.3.1 alphafold/2.3.2</pre><h3><span class="fontColorThemeSecondary">AlphaFold 2.3.2</span></h3><p><span class="fontColorThemeSecondary">We have added a plotting feature that runs at the end of the AlphaFold prediction job. The plots will be added to a directory called `plot` in your results directory. It will contain the following:</span></p><ul><li><span class="fontColorThemeSecondary">&nbsp;</span>MSA with Coverage</li><li>LDDT PER Position</li><li>Predicted LDDt and predicted aligned error&nbsp;per model (only if pTM models or multimer is used.)</li></ul>
<p>To load v2.3.2</p><pre>module load alphafold/2.3.2\nalphafold --help</pre>
<p>To test running AlphaFold from the command line, do the following:</p><ol><li>Make sure you have a vast scratch space or else change&nbsp;/vast/scratch/users/$USER/ to any other directory</li><li>The sample fasta file is here '<i><span class="fontColorBlue">/stornext/System/data/apps/sample-scripts/alphafold/scripts/2.3.2/rcsb_pdb_6MRR.fasta </span></i>'</li><li>Run the following command&nbsp;</li></ol><pre>alphafold -o /vast/scratch/users/$USER/AF_output -g  true -n 0,1 -t 2022-01-10 -r all /stornext/System/data/apps/sample-scripts/alphafold/scripts/2.3.2/rcsb_pdb_6MRR.fasta</pre><p style="margin-left:40px;"><br>4. The results will be created in a directory called '<i><span class="fontColorBlue">AF_output</span></i>' in your VAST scratch space.</p>
<p>AlphaFold2.3.2 can be run similarly to the<a href="/sites/rc2/SitePages/AlphaFold.aspx#running-alphafold2.0-on-milton-tutorial"> tutorial&nbsp;below</a>​​​​​​​ but use scripts from<strong> /stornext/System/data/apps/sample-scripts/alphafold/scripts/2.3.2</strong><i><span class="fontColorBlueLight">, </span></i>run ./run_alphafold_wf.sh and follow the command prompt.</p>
<h3><span class="fontColorThemeSecondary">Using Custom Templates</span></h3><p>You can use a custom templates path through the flag -c.</p><h3><span class="fontColorThemeSecondary">Using Custom Database</span></h3><p>&nbsp;</p><p>Use the following environment variables to use custom databases:</p><ul><li>AF_UNIREF30_DB: to override HHblits uniref30 database. It should be set to a directory containing UniRef30_YYYY_MM_* files.</li><li>AF_PDB70_DB: to override HHsearch pdb70 database. &nbsp;It should be set to a directory containing pdb70_* files.</li></ul><p>For example:</p><pre><br>#!/bin/bash<br><br>#SBATCH --time=24:00:00<br>#SBATCH --job-name=test<br>#SBATCH --mem 56G<br>#SBATCH -N 1<br>#SBATCH --cpus-per-task=4<br>#SBATCH -p regular<br>#SBATCH --export=ALL<br>#SBATCH -o log.txt<br><br><br>module purge<br>module load alphafold/2.3.2<br><br>today=$(date +%F)<br><br>export AF_UNIREF30_DB=/vast/projects/alphafold/databases/UniRef30_2023_02<br><br>alphafold -f -o /vast/scratch/users/esteva.m/alphafold/output_8OE6 -t ${today} -m monomer -i 5 /vast/scratch/users/$USER/8OE6.fasta</pre>
<p><span style="font-family:Arial, Helvetica, sans-serif;">If you get a warning message in your log file that is similar to the following:</span></p><p><span style="font-family:Arial, Helvetica, sans-serif;"><span class="fontColorRedDark"><strong>"WARNING: Number of match columns too large. Only first 19999 match columns will be kept!"</strong></span></span></p><p>Details are <a href="https://github.com/google-deepmind/alphafold/issues/810" data-interception="off" target="_blank" rel="noopener noreferrer"><strong>here</strong></a><strong>. </strong>Consider using the above environment variables to make use of the latest databases.</p>
<h3><span class="fontColorThemeSecondary">Prepare Protein sequence file</span></h3>
<p>Before running Alphafold 2, you need to have a .fasta file with your input sequence prepared.</p><p>For example, <span class="fontColorBlue">monomer</span> file will look like this</p><pre>&gt;sequence_name\n&lt;SEQUENCE&gt;</pre><p>A homomer with 3 copies of the same sequence &lt;SEQUENCE&gt;</p><pre>&gt;sequence_1\n&lt;SEQUENCE&gt;\n&gt;sequence_2\n&lt;SEQUENCE&gt;\n&gt;sequence_3\n&lt;SEQUENCE&gt;</pre><p>An A2B3 heteromer, i.e. with 2 copies of &lt;SEQUENCE A&gt; and 3 copies of &lt;SEQUENCE B&gt;</p><pre>&gt;sequence_1\n&lt;SEQUENCE A&gt;\n&gt;sequence_2\n&lt;SEQUENCE A&gt;\n&gt;sequence_3\n&lt;SEQUENCE B&gt;\n&gt;sequence_4\n&lt;SEQUENCE B&gt;\n&gt;sequence_5\n&lt;SEQUENCE B&gt;\n</pre><p>&nbsp;</p>
<p>To run on nextflow tower as described <a href="/sites/rc2/SitePages/Protein-Structure-Prediction-using-Seqera-Platform.aspx" data-interception="off" target="_blank" rel="noopener noreferrer"><strong>here</strong></a>, you will also need to know the absolute path where you have save your fasta files.&nbsp;</p><p>Both of the above steps can be done via Open Ondemand File Manager, as follows:</p>
<ol><li>Go to <a href="https://ondemand.hpc.wehi.edu.au/" data-interception="off" target="_blank" rel="noopener noreferrer">https://ondemand.hpc.wehi.edu.au/</a> and log in using your WEHI username and password</li><li><p>Go to File→ VAST, this will open your VAST scratch directory. If you do not have one, please request it here. You can choose to create your fasta files on your VAST project directory or any other directory.</p><p>&nbsp;</p></li></ol>
<p>3. You can create a new folder, using New Directory button And the use New File to create a new file *.fasta and add your sequence.</p>
<p>4. To get the full path to your file, use Copy Path.</p>
<h2><span class="fontColorThemeSecondary">AlphaFold v2.2.0.0</span></h2><p>AlphaFold 2.2.0.0 is the latest update from&nbsp;<a href="http://www.deepmind.com">DeepMind</a>&nbsp; and it is now available on the Milton modules system. Minor changes were done in this update.</p><ul><li>The AlphaFold-Multimer model weights have been updated, these new models have greatly reduced numbers of clashes on average and are slightly more accurate.</li><li>Adding a flag&nbsp;&nbsp;for&nbsp;how many predictions will be made per model, by default the offline system will run each model 5 times for a total of 25 predictions.</li><li>​​​​​​​The is_prokaryote_list&nbsp;flag has been removed,&nbsp;eukaryotes and prokaryotes are now paired in the same way.</li></ul><p>AlphaFold2.2.0.0&nbsp;can be run similar to the tutorial for 2.1 below but use scripts from<strong> </strong><i><span class="fontColorBlue"><strong>/stornext/System/data/apps/sample-scripts/alphafold/scripts/2.2</strong></span><span class="fontColorBlueLight">, </span></i>run ./run_alphafold_wf.sh and follow the command prompt.</p>
<h2><span class="fontColorThemeSecondary">AlphaFold v2.1.2</span></h2><p>AlphaFold 2.1.2 is an update from&nbsp;<a href="http://www.deepmind.com">DeepMind</a>&nbsp; and it is now available on the Milton modules system. Minor changes were done in this update.</p><ul><li>Adding a flag to escape relaxation step</li><li>Adding a flag to run relaxation on CPUs instead of GPUs</li></ul><p style="margin-left:40px;">&nbsp;</p><p>AlphaFold2.1.2 can be run similar to the tutorial for 2.1 below but use scripts from<strong> /stornext/System/data/apps/sample-scripts/alphafold/scripts/2.1.2</strong><i><span class="fontColorBlueLight">, </span></i>run ./run_alphafold_wf.sh and follow the command prompt.</p>
<h2><span class="fontColorThemeSecondary">AlphaFold v2.1</span></h2><p>AlphaFold 2.1 is the latest major update from&nbsp;<a href="http://www.deepmind.com">DeepMind</a>&nbsp;released on 4th November 2021. This version&nbsp;provides an implementation of AlphaFold-Multimer which is <strong>still&nbsp;a work in progress </strong>and AlphaFold-Multimer isn't expected to be as stable as the monomer AlphaFold system.</p><p>Alphafold2.1&nbsp; is available on the Milton modules system</p>
<h3><span class="fontColorThemeSecondary">Running AlphaFold2.1 on Milton tutorial</span></h3>
<p>AlphaFold2.1 can be run similar to the tutorial below but use scripts from<strong> </strong><i><span class="fontColorBlue"><strong>/stornext/System/data/apps/sample-scripts/alphafold/scripts/2.1</strong></span><span class="fontColorBlueLight">, </span></i>run ./run_alphafold_wf.sh and follow the command prompt, it will be different to the example below. So make sure you read the questions well. A detailed tutorial will be added soon.</p><p>Preparing input sequence files for monomers is the same but for mutimers:</p><p>To fold a&nbsp;homomer, for example a homomer with 3 copies</p><pre>&gt;sequence_1\n&lt;SEQUENCE&gt;\n&gt;sequence_2\n&lt;SEQUENCE&gt;\n&gt;sequence_3\n&lt;SEQUENCE&gt;</pre><p>To fold a multimer,&nbsp;for example a&nbsp;heteromer with 2 copies of&nbsp;<i><span class="fontColorBlue">&lt;SEQUENCE A&gt;</span></i><span class="fontColorBlueLight">&nbsp;</span>and 3 copies of&nbsp;<i><span class="fontColorBlue">&lt;SEQUENCE B&gt;</span></i>.</p><pre>&gt;sequence_1\n&lt;SEQUENCE A&gt;\n&gt;sequence_2\n&lt;SEQUENCE A&gt;\n&gt;sequence_3\n&lt;SEQUENCE B&gt;\n&gt;sequence_4\n&lt;SEQUENCE B&gt;\n&gt;sequence_5\n&lt;SEQUENCE B&gt;\n</pre>
<h2><span class="fontColorThemeSecondary">AlphaFold v2.0</span></h2><p>Alphafold v2.0, developed by <a href="http://www.deepmind.com" data-interception="off" target="_blank" rel="noopener noreferrer">DeepMind</a>, is a completely new model that was entered in <a href="https://www.predictioncenter.org/casp14/index.cgi" data-interception="off" target="_blank" rel="noopener noreferrer">CASP14</a> and published in this article&nbsp;'<a href="https://www.nature.com/articles/s41586-021-03819-2" data-interception="off" target="_blank" rel="noopener noreferrer">Highly accurate protein structure prediction with AlphaFold</a>' in Nature magazine. The source code can be found in <a href="https://github.com/deepmind/alphafold" data-interception="off" target="_blank" rel="noopener noreferrer">this repo</a>.</p><p>Alphafold2.0 is available on the Milton modules system</p>
<p>The image on the right shows a comparison between a reconstruction of <a href="https://www.rcsb.org/structure/6Y4F" data-interception="off" target="_blank" rel="noopener noreferrer">6Y4F</a> using x-ray (red) and AlphaFold (green) produced on WEHI Milton GPU nodes.</p><pre>module av alphafold\n\n--------------- /stornext/System/data/modulefiles/structbio/its ----------------\nalphafold/2.0.0(default) alphafold/2.0.2\nalphafold/2.0.1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;alphafold/2.1.1.0\n\n</pre><p>We have three versions of AlphaFold2.0, they are similar. However, with alphafold/2.0.1 you can run the preprocessing step independent of the deep learning inference step.</p><p>This is more efficient as the preprocessing step requires CPUs only and thus can run efficiently and more rapidly on the regular partition.</p><p>​​​​​​​alphafold/2.0.2 has the option to run homo-oligomers. The homo-oligomer option&nbsp;is based on implementation&nbsp;presented&nbsp;<a href="https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb#scrollTo=33g5IIegij5R">here</a>&nbsp;by&nbsp;<strong>Minkyung&nbsp;Baek&nbsp;</strong><a href="https://twitter.com/minkbaek">(@minkbaek</a>) and <strong>Yoshitaka&nbsp;Moriwaki</strong>&nbsp;(<a href="https://twitter.com/Ag_smith">@Ag_smith</a>).&nbsp;&nbsp;</p><p>Please bear in mind, that&nbsp;these are experimental&nbsp;options.&nbsp;</p>
<h3><span class="fontColorThemeSecondary">Running AlphaFold2.0 on Milton tutorial</span></h3>

<ol><li>Login to WEHI: Open&nbsp;Ondemand&nbsp;(<a href="https://ondemand.hpc.wehi.edu.au/">https://ondemand.hpc.wehi.edu.au/</a>)&nbsp;<ol><li>You need to be connected to the WEHI network by using your WEHI computer onsite or through the VPN. If you are accessing Milton from a non-WEHI device, you need to log in via the remote access portal (<a href="https://rap.wehi.edu.au/">https://rap.wehi.edu.au/</a>). &nbsp;</li><li>If this is your first time, you need to request access to Slurm by emailing&nbsp;<a href="mailto:support@wehi.edu.au" data-interception="off" target="_blank" rel="noopener noreferrer">support@wehi.edu.au</a>.&nbsp;&nbsp;</li><li>The username and password will be the same as your WEHI username and password. ​​​​​​​<br><br>&nbsp;</li></ol></li><li>Using the Files Menu (Figure 2), after logging in to Ondemand, choose where you want to create your&nbsp;fasta&nbsp;file and get your output. We recommend using VAST Scratch for that. VAST is fast&nbsp;but REMEMBER to copy the results to another area once the jobs are completed to avoid loss if they are not accessed in 14 days. If you do not have access to VAST, request access through the Helpdesk. In this tutorial, VAST is used (Figure 3).<br><br>&nbsp;</li><li>Create a folder for the project (Figure 4) for example, called AF, and navigate to the folder (Figure 5)​​​​​​​.</li></ol>
<p aria-hidden="true">&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-captiontext="Figure 2" data-imagenaturalheight="118" data-imagenaturalwidth="468" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/1397987651-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="9481455c-129a-4fe2-a5c3-f4beb404c113" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="78.3948667324778" data-width="310.9220138203357" data-widthpercentage="92.39881539980259"></div><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-captiontext="Figure 3" data-imagenaturalheight="78" data-imagenaturalwidth="433" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/3675200357-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="0d724103-7120-46bc-bfdb-ad5472aee55f" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="51.82033563672262" data-width="287.6692991115499" data-widthpercentage="85.48864758144128"></div><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-captiontext="Figure 4" data-imagenaturalheight="195" data-imagenaturalwidth="364" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/1282496631-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="1bf09fec-2239-4579-8359-699f3b71d5bb" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="129.55083909180652" data-width="241.82823297137216" data-widthpercentage="71.86574531095755"></div><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-captiontext="Figure 5" data-imagenaturalheight="100" data-imagenaturalwidth="335" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/4007069161-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="cad78324-d797-4f9f-a386-4d7a0731f677" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="66.43632773938796" data-width="222.56169792694968" data-widthpercentage="66.14017769002962"></div><p>&nbsp;</p>
<p>4. Prepare your&nbsp;fasta&nbsp;file&nbsp;</p><p style="margin-left:40px;">a. Create a new file (query.fasta)&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="132" data-imagenaturalwidth="339" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/2528398320-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="34f1027b-7a79-458e-91cc-1f24fb05504d" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="87.6959526159921" data-width="225.21915103652518" data-widthpercentage="66.92991115498519"></div><p style="margin-left:40px;">b. Edit the file&nbsp;&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="120" data-imagenaturalwidth="323" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/1374747399-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="ad3e1f27-935a-4e22-8782-abe27805d6e0" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="79.72359328726553" data-width="214.58933859822304" data-widthpercentage="63.770977295162865"></div><p style="margin-left:40px;">c. Add the protein and save&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="112" data-imagenaturalwidth="395" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/4090909848-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="3d1a1fc7-7252-40a7-8cd7-18106ef12d90" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="74.40868706811452" data-width="262.4234945705824" data-widthpercentage="77.98617966436328"></div><p>&nbsp;5. Copy scripts from&nbsp;<i><span class="fontColorBlue"><strong>/stornext/System/data/apps/sample-scripts/alphafold/scripts/2.0</strong></span></i>&nbsp;to your folder&nbsp;</p><p style="margin-left:40px;">a. Change Directory to&nbsp;<i><span class="fontColorBlue">/stornext/System/data/apps/sample-scripts/alphafold/scripts/2.0</span></i></p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="91" data-imagenaturalwidth="336" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/448670854-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="7abb4024-5172-4c3e-ac06-76ffe85dbd6c" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="60.45705824284303" data-width="223.22606120434352" data-widthpercentage="66.33761105626851"></div><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="84" data-imagenaturalwidth="136" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/2919062397-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="ef779c15-021a-4346-afa5-0cf3ce741b8c" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="55.80651530108588" data-width="90.35340572556761" data-widthpercentage="26.850937808489633"></div><p>&nbsp;</p><p style="margin-left:40px;">&nbsp;</p>
<p style="margin-left:40px;">&nbsp;b.&nbsp;Choose all files there and click on Copy/Move</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="160" data-imagenaturalwidth="436" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/1872727615-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="58d01fa3-af84-428e-82a4-2bffd339ae3c" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="106.29812438302075" data-width="289.6623889437315" data-widthpercentage="86.08094768015795"></div><p style="margin-left:40px;">c.&nbsp;Navigate back to your folder and Copy​​​​​​​</p><p style="margin-left:40px;">&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="163" data-imagenaturalwidth="435" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/3993225550-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="ae69cdc3-f5b9-45d4-8c6b-b288ccf2db2a" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="108.29121421520236" data-width="288.9980256663376" data-widthpercentage="85.88351431391904"></div><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="143" data-imagenaturalwidth="432" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/4122142541-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="9fd8bbec-7446-4ae7-b250-3f4aa1ec09d1" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="95.00394866732478" data-width="287.00493583415596" data-widthpercentage="85.29121421520236"></div><p style="margin-left:8px;">6. Click on “Open In Terminal”, a new tab will appear. You will be asked for your password if you have no keys set.</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="37" data-imagenaturalwidth="334" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/596674984-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="1e43e0ba-f827-4fc2-9a6e-fc9091da54fe" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="24.581441263573545" data-width="221.89733464955577" data-widthpercentage="65.94274432379072"></div><p style="margin-left:8px;">7.&nbsp;Run the command ./run_alphafold_wf.sh.&nbsp;Below shows the result of running the command, the <span class="highlightColorYellow">highlighted</span> parts are user inputs.&nbsp;</p><p style="margin-left:8px;">&nbsp;</p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="157" data-imagenaturalwidth="556" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/3173022062-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="c5ee46ea-a274-4822-a73d-b38a4030c125" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="95.01888489208633" data-width="336.5" data-widthpercentage="100"></div><p>&nbsp;</p>
<pre>./run_alphafold_wf.sh  \nOutput directory [here]:<span class="highlightColorYellow">./</span> \nFound the following fasta files \nquery.fasta \nDo you wish to run preprocessing pipeline? [y]/n: <span class="highlightColorYellow">y</span> \nDo you wish to run model inference pipeline? [y]/n: <span class="highlightColorYellow">y</span> \nHow many models? [1]<span class="highlightColorYellow">5</span> \nDo you wish to run on gpus? [y]/n: <span class="highlightColorYellow">y</span> \nHow many homo-oligomers to use? [1]<span class="highlightColorYellow">2</span> \nOutput directory : /vast/scratch/users/iskander.j/AF \nSubmited preprocessing job 2134442 \nSubmited model model_1 job 2134443 \nSubmited model model_2 job 2134444 \nSubmited model model_3 job 2134445 \nSubmited model model_4 job 2134446 \nSubmited model model_5 job 2134447 </pre>
<p>The&nbsp;run_alphafold_wf.sh was initially written by <span class="fontColorBlue"><strong>Josh Hardy</strong></span> and then modified to include homooligomers and cpu-only option by <span class="fontColorBlue"><strong>Julie Iskander</strong></span></p>
<pre>Output directory [here]: </pre><p>User should input the path to the output directory. Since we opened the script in the directory that contains all scripts and the&nbsp;fasta&nbsp;file, you can just type&nbsp; <span class="fontColorBlue"><strong>./&nbsp;</strong></span> which means the current directory.</p><pre>Do you wish to run preprocessing pipeline? </pre><p>This is just the first part shown in&nbsp;Figure&nbsp;1. This will run only on CPUs.&nbsp;</p><pre>Do you wish to run model inference pipeline? </pre><p>This is the&nbsp;second&nbsp;part shown in&nbsp;Figure&nbsp;1. This&nbsp;can&nbsp;run on GPUs and CPUs. GPUs&nbsp;will be faster, but long proteins may run out of GPU memory, if this happens rerun using only&nbsp;cpus, as will be shown next.</p><pre>How many models? [1] </pre><p>AlphaFold&nbsp;has 5 different trained deep&nbsp;learning models specified that can be used. So user can choose to run against only one or more models, maximum 5.&nbsp;</p><pre>Do you wish to run on gpus? [y]/n:  </pre><p>This is where user can choose to opt out of using&nbsp;gpus&nbsp;and run all the model inference step on&nbsp;cpus&nbsp;only, which will take longer.&nbsp;</p><pre>How many homo-oligomers to use? [1] </pre><p>Number of oligomers to&nbsp;create, if&nbsp;nothing is given default is 1, the maximum is 8. By trial more&nbsp;will&nbsp;create overlapping chains.&nbsp;</p>
<p><span class="fontColorBlue">Note that our GPU nodes have a memory of&nbsp;128GB per GPU. So long proteins might not&nbsp;successfully run. If you have a protein longer than 1.5k, we recommend using CPUs.</span>&nbsp;</p><p><span class="fontColorBlue"><strong>One GPU</strong> is recommended and a maximum of 10 CPU cores. The amount of memory depends on the input sequence, but remember that the GPU nodes only have ~128GB memory.</span></p>
<h3><span class="fontColorThemeSecondary">Output directory structure&nbsp;</span></h3><p>The results will eventually appear in your directory.&nbsp;For each&nbsp;Fasta&nbsp;file you had in the directory a&nbsp;new directory will be created with the following structure&nbsp;</p><p><span class="fontColorBlue"><strong>query/&nbsp;</strong></span></p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">features.pkl</span></i>&nbsp;A pickle file containing the input feature NumPy arrays used by&nbsp;the models&nbsp;to produce the structures.&nbsp;</p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">re-ranking_debug.json</span></i>&nbsp;A JSON format text file containing the&nbsp;pLDDT&nbsp;values used to perform the model ranking, and mapping back to the original model names.&nbsp;</p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">relaxed_model_{1,2,3,4,5}.pdb</span></i>&nbsp; A PDB format text file containing the predicted structure, after performing an Amber relaxation procedure on the unrelaxed structure prediction (see Jumper et al. 2021, Suppl. Methods 1.8.6 for details).&nbsp;</p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">relaxed_model_{1,2,3,4,5}_reset.pdb</span></i> Same as above but with added&nbsp;pLDDT&nbsp;as&nbsp;bfactor&nbsp;and&nbsp;if more than one oligomer, the chain id is reset to show different chains.&nbsp;</p><p style="margin-left:40px;">|--<i><span class="fontColorBlue">&nbsp;result_model_{1,2,3,4,5}.pkl</span></i>&nbsp; A pickle file containing a nested dictionary of the various NumPy arrays directly produced by the model. In addition to the output of the structure module, this includes auxiliary outputs.&nbsp;</p><p style="margin-left:40px;">|--&nbsp;<span class="fontColorBlue">unrelaxed_model_{1,2,3,4,5}.pdb&nbsp;</span></p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">unrelaxed_model_{1,2,3,4,5}_reset.pdb&nbsp;</span></i></p><p style="margin-left:40px;">|--&nbsp;<span class="fontColorBlue"><strong>msas/</strong></span>&nbsp; A directory containing the files describing the various genetic tool hits that were used to construct the input MSA&nbsp;and templates.&nbsp;</p><p style="margin-left:40px;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |--&nbsp;<i><span class="fontColorBlue">bfd_uniclust_hits.a3m&nbsp;</span></i></p><p style="margin-left:40px;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|--&nbsp;<i><span class="fontColorBlue">mgnify_hits.sto&nbsp;</span></i></p><p style="margin-left:40px;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|--&nbsp;<i><span class="fontColorBlue">uniref90_hits.sto&nbsp;</span></i></p><p style="margin-left:40px;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |--&nbsp;<i><span class="fontColorBlue">pdb70_hits.hhr&nbsp;</span></i></p><p>If&nbsp;homooligomer&nbsp;value&nbsp;used is greater than 1, a few more files will exist.&nbsp;</p><p style="margin-left:40px;">|--<i><span class="fontColorBlue"> chains_h{value}.pdb</span></i>&nbsp;&nbsp;Chain&nbsp;id to be used to reset&nbsp;chains in predicted structures&nbsp;</p><p style="margin-left:40px;">|--&nbsp;<i><span class="fontColorBlue">features_h{value}.pkl</span></i>&nbsp;same as&nbsp;features.pkl&nbsp;but&nbsp;msas&nbsp;and residues index are modified to allow oligomers.&nbsp;</p><p>There may be other files, but they are not useful, due to the modifications done.&nbsp;&nbsp;</p>
<p>If the model inference jobs fail due to out of memory,&nbsp;you can change the&nbsp;associated&nbsp;SLURM directives at the top of the&nbsp;sbatch&nbsp;jobs&nbsp;​​​​​​​</p><ul><li>alphafold-h-step2_cpu.sbatch,&nbsp;when using CPUs only</li></ul><pre>#SBATCH --time=48:00:00\n#SBATCH --job-name=AF2_cpu\n#SBATCH --mem 100G\n#SBATCH --cpus-per-task=10\n#SBATCH --export=ALL\n#SBATCH -o logs/%x_%j.out​​​​​​​\n</pre><ul><li>​​​​​​​alphafold-h-step2_gpu.sbatch è when using GPUs</li></ul><pre>#SBATCH --time=48:00:00\n#SBATCH --job-name=AF2_gpu\n#SBATCH --mem <span class="highlightColorYellow">30G</span>\n#SBATCH --cpus-per-task=10\n#SBATCH -p gpuq\n#SBATCH --gres=gpu:1\n#SBATCH --export=ALL\n#SBATCH -o logs/%x_%j.out</pre><p style="margin-left:96px;">&nbsp;</p>
<h3><span class="fontColorThemeSecondary">Amber relaxation and homo-oligomers</span></h3><p>Currently, amber relaxation for homo-oligomers cannot process multiple chains and will force linkages between chain breaks.&nbsp;Therefore, it is not&nbsp;performed when you choose homooligomers &gt;1 (alphafold/2.0.2).</p><p>It is recommended to relax the structure in another program using the unrelaxed_model_{x}_reset.pdb file e.g. <a href="http://www.phenix-online.org/documentation/reference/geometry_minimization.html" data-interception="off" target="_blank" rel="noopener noreferrer"><strong>PHENIX geometry minimisation&nbsp;</strong></a></p><div class="imagePlugin" style="background-color:transparent;position:relative;" data-alignment="Center" data-imagenaturalheight="295" data-imagenaturalwidth="360" data-imageurl="https://wehieduau.sharepoint.com/sites/rc2/SiteAssets/SitePages/AlphaFold/4076244984-image.png" data-isdecorative="false" data-listid="3d75202a-c480-42f5-abb7-5e7b39fc27d5" data-siteid="93ee3842-f21b-4d8d-b85b-874e64c67802" data-uniqueid="c6a2ef8c-3da8-41d8-9a3b-7b23ba8be00a" data-webid="95711e2c-a16f-49a8-9adc-1b9756d4ee5e" data-height="200.82159624413146" data-width="245.07042253521126" data-widthpercentage="33.80281690140845"></div><p>​​​​​​​</p>
<h3>AlphaFold command</h3><p>To load AlphaFold:</p><pre>module load alphafold/2.0.1\n</pre><pre>alphafold -h\n\nUsage: alphafold [OPTION: -g -l -d -p -f -m -b -h] -o &lt;output path&gt; -t &lt;max template date&gt; &lt;input FASTA file(s)&gt;...\n\nAlphaFold Singularity wrapper - Miguel Esteva (esteva.m@wehi.edu.au)\n\nSingularity is needed to run this script!\n\nPlease ensure each input FASTA has one sequence only.\n\nOPTION:\n  -o  output path         [Required] Path to a directory that will store the results. Will be created if non-existent.\n  -g  use GPU             Enable (true)/disable (false) the use of GPU (Default: true).\n  -l  GPU(s) to use       Comma separated list of GPU(s) to use e.g. 0,1,2,3. Will be ignored in Slurm.\n  -d  databases path      Path to the directory containing relevant databases. (Default: /vast/projects/alphafold/databases).\n  -p  preset              Choose preset model configuration: reduced_dbs, full_dbs or casp14. (Default: full_dbs).\n  -f  features only       Stop pipeline once feature extraction is complete and features.pkl is created.\n  -m  model names         Model list to use (comma-separated list model_1,...,model_5. No spaces). (Default: model_1).\n  -b  benchmark mode      Enable benchmark mode by setting -b. (Default: disabled).\n                              Run multiple JAX model evaluations to obtain a timing that excludes\n                              the compilation time, which should be more indicative.\n                              of the time required for inferencing many proteins.\n  -t  max template date   [Required] Maximum template release date to consider in YYYY-MM-DD format.\n  -h  help                Print this help menu.\n\nFor issues with this script, please contact Helpdesk.\nFor issues with AlphaFold, please refer to: https://github.com/deepmind/alphafold/issues\n\n</pre><p>To run it, first prepare your input FASTA file and ensure that it has <strong>one sequence only</strong>.</p><p>To run the preprocessing only:</p><pre>alphafold -o output -t 2020-05-14 -m model_1 -f &lt;myfasta.fasta&gt;\n</pre><p>Once it is finished, you can run the GPU-based processing part:</p><pre>alphafold -o output -t 2020-05-14 -m model_1 query.fasta\n</pre><p>To run both steps in one command:</p><pre>module load alphafold/2.0.0\nalphafold -o output -t 2020-05-14 -m model_1 &lt;myfasta.fasta&gt;\n</pre><p>output is the directory where the output files will be saved and <i><span class="fontColorBlue">myfasta.fasta</span></i> is the filepath to your fasta sequence input file. Remember that Alphafold needs GPU to run efficiently, however it can also run without GPUs but the runs will be 10x slower.</p><p><i><span class="fontColorBlue">alphafold.slurm</span></i></p><pre>#!/bin/bash\n\n#SBATCH --time=12:00:00\n#SBATCH --job-name=AF2\n#SBATCH --mem=10G\n#SBATCH --cpus-per-task=10\n#SBATCH -p gpuq\n#SBATCH --gres=gpu:1\n#SBATCH --export=ALL\n#SBATCH -o %x_%j.out\n#SBATCH -e %x_%j.err\n\nmodule load alphafold\nalphafold -o output -t 2020-05-14 -m model_1 query.fasta\n\n</pre><p>To submit a job into the gpuq:</p><pre>sbatch alphafold_v2.0.1.slurm\n</pre><p>Once the job starts two files in the current directory will be created <i><span class="fontColorBlue">AF_&lt;jobid&gt;.out</span></i> and <i><span class="fontColorBlue">AF_&lt;jobid&gt;.err</span></i>, to help you monitor the progress of your job and if any error had happened. Always keep an eye on these two files.</p>
<h3><span class="fontColorThemeSecondary">Discussion group meeting</span></h3><p>The group met on&nbsp; 13<sup>th</sup> September 2021&nbsp;and agreed on the following points:</p><ol><li><span class="fontColorBlue">Update&nbsp;Documentation&nbsp;with how to&nbsp;</span><ol><li><span class="fontColorBlue">prepare files,&nbsp;&nbsp;</span></li><li><span class="fontColorBlue">run AF&nbsp;</span></li><li><span class="fontColorBlue">run on&nbsp;CPUs&nbsp;only&nbsp;</span></li></ol></li><li>Organise&nbsp;AlphaFold training session&nbsp;</li><li><span class="fontColorBlue">Add support to&nbsp;run&nbsp;homo-oligomers</span>&nbsp;and&nbsp;complexes</li><li>Develop an&nbsp;easy-to-use&nbsp;interface to run AF2&nbsp;</li><li><span class="fontColorBlue">Searching AF predicted database locally.&nbsp;</span>​​​​​​​</li></ol>
<h3>More Resources</h3>
