<h2>Dalilite</h2><p><a href="/sites/rc2/SitePages/Dali-Search.aspx#dalilite" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dalilite" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dalilite">Dalilite tool</a> is now available on <strong>Milton</strong> and the local store of&nbsp;<a href="/sites/rc2/SitePages/AlphaFold.aspx" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/AlphaFold.aspx" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/AlphaFold.aspx">AlphaFold database</a> has been archived and removed as of September 2023.&nbsp;&nbsp;</p>
<p>There is also an online&nbsp;<a href="http://ekhidna2.biocenter.helsinki.fi/dali/" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/" target="_blank" data-interception="off" title="http://ekhidna2.biocenter.helsinki.fi/dali/">Dali Protein Structure Comparion Server&nbsp;</a>&nbsp;which is a network service for comparing protein structures in 3D. You submit the coordinates of a query protein structure and Dali compares them against those in the Protein Data Bank (PDB).&nbsp;</p>
<h4><span class="fontColorBlue">Contents on this page:</span></h4><p><a href="/sites/rc2/SitePages/Dali-Search.aspx#dalilite" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dalilite" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dalilite">Dalilite</a><br><a href="/sites/rc2/SitePages/Dali-Search.aspx#import.pl" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#import.pl" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#import.pl">Import.pl</a><br><a href="/sites/rc2/SitePages/Dali-Search.aspx#dali.pl" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dali.pl" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#dali.pl">Dali.pl</a><br><a href="/sites/rc2/SitePages/Dali-Search.aspx#create_dali_summary.py" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#create_dali_summary.py" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Dali-Search.aspx#create_dali_summary.py">Create dali_summary.py</a><br></p>
<h3><span class="fontColorThemeSecondary">Dalilite</span></h3><p><a title="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" data-interception="off" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" target="_blank">Dalilite</a>&nbsp;is a tool for protein structural alignment and structure database search. It is based on the Dali method (Dali stands for <strong>d</strong>istance matrix <strong>ali</strong>gnment). It consists of two wrapper scripts:<br><br></p><ul><li><a href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#import" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#import">import.pl</a><a title="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#import" data-interception="on" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#import" href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#import">&nbsp;</a>- convert PDB files to internal data format used by Dali, mirror PDB database</li><li><a href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#dali5" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#dali5">dali.pl</a><a title="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#dali5" data-interception="on" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#dali5" href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html#dali5">&nbsp;</a>- perform pairwise structural alignments, all-against-all comparisons, or structural database searches</li></ul><p>Dalilite is now available on Milton through the modules system. The installed version&nbsp;supports parallelisation through mpi.</p><pre>module av dalilite\n\n--------------- /stornext/System/data/modulefiles/structbio/its ----------------\ndalilite/5.1(default)\n\n\n\nmodule help dalilite\n\n----------- Module Specific Help for 'dalilite/5.1' ---------------\n\n    This module sets the environment for dalilite v5.1\n\n    dali.pl -h\n    import.pl -h</pre><p>To load it:&nbsp;</p><pre>module load dalilite\n</pre>
<h4><span class="fontColorThemeSecondary">import.pl</span></h4><p><em><span class="fontColorBlue">import.pl</span></em> converts a PDB entry into the internal data format used by Dali. It reads files in PDB format and creates the corresponding data files, one for each chain, in the data directory. PDB files can be imported one at a time.&nbsp;</p><p>For example:</p><p style="margin-left:40px;"><span class="fontColorBlue">import.pl --pdbfile &lt;filename&gt; --pdbid &lt;xxxx&gt; [ --dat &lt;path&gt; ]</span></p><p><strong>Options:</strong></p><pre>      --dat &lt;path&gt;                    directory to store imported data [default: ./DAT]\n        --pdbfile &lt;filename&gt;       PDB formatted file, may be compressed (.gz). &lt;filename&gt; includes path\n        --pdbid &lt;xxxx&gt;                four-letter PDB identifier\n        --pdblist &lt;filename&gt;       list of PDB entries. &lt;filename&gt; includes path and must match the pattern pdb????.ent\n        --rsync                              automated PDB mirroring\n        --pdbmirrordir &lt;path&gt;    PDB mirror directory [default: /data/pdb]\n        --clean                              remove temporary files\n        --verbose                          verbose</pre>
<h4><span class="fontColorThemeSecondary">dali.pl</span></h4><p><em><span class="fontColorBlue">dali.pl</span></em> performs pairwise structural alignment and database searches. The full list of options and syntax can be found <a title="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" data-interception="off" data-cke-saved-href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" href="http://ekhidna2.biocenter.helsinki.fi/dali/README.v5.html" target="_blank">here</a><strong>.</strong></p><p>For example:</p><pre>dali.pl  --MPIRUN_EXE mpirun --np 2\t \\\n                                         --cd1 &lt;xxxxx&gt; \\\n                                         --db &lt;filename&gt; \\\n                                         --TITLE &lt;string&gt;  \\\n                                         --dat1 &lt;path&gt;  \\\n                                         --dat2 &lt;path&gt; \\\n                                         --outfmt summary,alignments,equivalences,transrot</pre><p><strong>Options:</strong>&nbsp; &nbsp;</p><pre style="margin-left:40px;">--cd1 &lt;xxxxX&gt;             query structure identifier       \n--db &lt;filename&gt;          list of target structure identifiers\n\n--dat1 &lt;path&gt;              path to directory containing query data [default: ./DAT/]\n--dat2 &lt;path&gt;             path to directory containing target data [default: ./DAT/]\n\n--title &lt;string&gt;            written to output [default: test]\n\n--outfmt &lt;string&gt;       result blocks to output: summary,alignments,equivalences,transrot [default: summary]\n--clean                         remove temporary files\n\n--np &lt;integer&gt;            number of processes if using openmpi (between 1 and 99) [default: 1]\n</pre><p><br></p>
<h4><span class="fontColorThemeSecondary">create_dali_summary.py</span></h4><p><em><span class="fontColorBlue">create_dali_summary.py </span></em>&nbsp;is a python script created by <strong>Richard Birkinshaw</strong> and <strong>Julie Iskander </strong>that reformats the dali search summary output, adds extra columns, and resaves in csv format.&nbsp;</p><p>The csv file has these 14 columns:<br><br></p><ul><li>No,</li><li>Chain,</li><li>Z,</li><li>rmsd,</li><li>lali,</li><li>nres,</li><li>id,</li><li>PDB Description,</li><li>ReferenceProteome,</li><li>DaliPDBCode,</li><li>AlphafoldPDB,</li><li>UniProtID,</li><li>species,</li><li>common_name</li></ul><p>The file is saved in the path set by option <em><span class="fontColorBlue">--output_dir</span></em> or the current directory if that option is not set.</p><p>For example:</p><pre>create_dali_summary.py  --output_dir &lt;path to dali output directory&gt;\n</pre><p><br></p>
<p><em><span class="fontColorBlue">DaliLite writes a number of intermediate results in the current working directory (CWD). The lock file is deleted automatically if the job is completed successfully. If a file named <strong>dali.lock</strong> is present, you get the following error message and cannot start another DaliLite job in the same directory:</span></em></p><pre>Directory is locked by dali.lock\n        there may be another DALI process running in this work directory\n         or, the previous run crashed: remove the dali.lock file</pre>
<h3><span class="fontColorThemeSecondary">AlphaFold prediction database has been <span class="fontColorRedDark">a</span></span><span class="fontColorRedDark"><span class="fontColorRedDark">rch</span>ived and is no longer available as of SEPTEMBER 2023.</span></h3>
