<h2>Accessing software</h2><h3><span class="fontColorThemeSecondary">Modules system</span></h3><p>All application software on Milton nodes is provided through the modules system: <a data-cke-saved-href="http://modules.sourceforge.net/" href="http://modules.sourceforge.net/" target="_blank" title="http://modules.sourceforge.net/" data-interception="off">modules.sourceforge.net/</a></p><p>The modules system configures the user's unix environment to run specific applications, or indeed specific versions of applications, and to avoid conflicts from mutually exclusive software. The unix environment here refers to environment variables​ such as the ubiquitous <em><span class="fontColorBlue">$PATH</span></em>, and also application-specific environment variables such as <span class="fontColorBlue"><em>$PYTHONPATH</em></span>.</p><p>The modules system is accessed via the <em><span class="fontColorBlue">module</span></em>​&nbsp;command. To list available software, log in to the unix system you wish to use:</p><pre>$ <span class="fontColorRed">module av</span>\n\n------------------------------------- /stornext/System/data/modulefiles/bioinf/its -------------------------------------\n7zip/15.14.1(default)             hdf5/1.8.16(default)              python/2.7.12(default)\nanaconda2/4.0.0(default)          hdfview/2.11.0(default)           python/3.5.1\nanaconda2/4.2.0                   hisat2/2.0.3                      python/3.5.3\nanaconda3/4.0.0(default)          hisat2/2.0.5(default)             R/3.2.5\nanaconda3/4.3.1                   homer/4.8.2(default)              R/3.3.0\nannovar/2015-12-14(default)       htslib/1.3.1                      R/3.3.2\napache-ant/1.9.7(default)         htslib/1.3.2                      R/3.3.3\nbamtools/2.4.1(default)           htslib/1.4.1                      R/3.4.0(default)\nbcftools/1.3.1(default)           htslib/1.5(default)               root/6.10.00(default)\nbeast/1.8.3(default)              igv/2.3.72(default)               ruby/2.3.1(default)\nbeast2/2.4.0(default)             igvtools/2.3.72(default)          sambamba/0.6.6(default)\nbedops/2.4.26(default)            IRFinder/1.2.0(default)           samblaster/0.1.23(default)\nbedtools/2.25.0                   java/1.7.0_79                     samtools/0.1.18\nbedtools/2.26.0(default)          java/1.8.0_131(default)           samtools/0.1.19\nbismark/0.16.1(default)           java/1.8.0_92                     samtools/1.3.1\nboost/1.55.0                      juncBASE/1.2-beta(default)        samtools/1.4.1\nboost/1.60.0(default)             kraken/0.10.5-beta(default)       samtools/1.5(default)\nbowtie/1.1.2(default)             last/864(default)                 scala/2.12.2(default)\nbowtie2/2.2.9(default)            macs2/2.1.0(default)              singularity/2.2(default)\nbpipe/0.9.9.2(default)            mpich/3.2(default)                singularity/2.3\nbreakdancer/1.1.2                 multiSNV/2.3-10-g06102a5(default) snpEff/4.3i(default)\nbreakdancer/1.4.5(default)        muscle/3.8.31(default)            SNPsplit/0.2.0(default)\nbwa/0.7.13                        mutect/1.1.7(default)             soapdenovo2/2.04(default)\nbwa/0.7.15(default)               nanonet/2.0.0(default)            soapdenovo-trans/1.03(default)\nbzip2/1.0.6(default)              nanonet-openBLAS/2.0.0(default)   sra-toolkit/2.8.2-1(default)\ncanu/1.3(default)                 nanopolish/0.5.0                  sshpass/1.05(default)\ncanu/1.5                          nanopolish/0.6.2(default)         STAR/2.5.2(default)\ncircos/0.69(default)              nanopore/1.0(default)             subread/1.5.0-p2\ncnvnator/0.3.3(default)           ncbi-blast/2.6.0+(default)        subread/1.5.1\ncutadapt/1.13(default)            nedit/5.5(default)                subread/1.5.2(default)\nensembl-vep/88.0                  netcdf/4.4.1.1(default)           texlive/3.14159265(default)\nensembl-vep/89.0(default)         node/6.11.0(default)              tophat/2.1.1(default)\nfastqc/0.11.5(default)            novoalign/3.04.04(default)        trimgalore/0.4.2\nflash/1.2.11(default)             oases/0.2.09(default)             trimgalore/0.4.4(default)\ngatk/3.5.0                        openBLAS/0.2.19(default)          trimmomatic/0.36(default)\ngatk/3.6.0                        openmpi/2.1.0(default)            trinityrnaseq/2.2.0\ngatk/3.7.0(default)               openmpi/2.1.1                     trinityrnaseq/2.4.0(default)\ngcc/6.3.0(default)                openmpi/2.1.1-java8_131           trinotate/3.0.1(default)\ngcc/7.1.0                         pandoc/1.17.0.3(default)          ucsc-tools/331(default)\ngenetorrent/3.8.7(default)        parallel/20161122(default)        usearch/8.1.1861\ngermline/1.5.1(default)           pbs-drmaa/1.0.19(default)         usearch/9.2.64(default)\ngmap-gsnap/2017-05-08             pcre2/10.21(default)              varscan/2.3.9(default)\ngmap-gsnap/2017-06-16(default)    perl/5.22.2(default)              vcflib/1.0.0_rc1(default)\ngraphmap/0.3.1                    php/5.6.21(default)               vcftools/0.1.13(default)\ngraphmap/0.5.1(default)           picard-tools/2.2.4                velvet/1.2.10(default)\ngridss/0.10.1(default)            picard-tools/2.9.4(default)       vsearch/2.4.3(default)\nhaplopainter/1.043(default)       pindel/0.2.5b9(default)           xz/5.2.2(default)\nhdf4/4.2.11(default)              poretools/0.6.0(default)          zlib/1.2.8(default)\n\n-------------------------------------------- /usr/share/Modules/modulefiles --------------------------------------------\ndot         module-git  module-info modules     null        use.own\n</pre><p><br></p><p>This list is an example only. New software is being added on almost a daily basis and the precise list of available software will vary from day to day. The subcommand <em><span class="fontColorBlue">av</span></em> is short for <span class="fontColorBlue"><em>avail</em></span> which can also be used.</p><p>A module is then loaded to make it available for use:</p><pre>$ module load samtools\n</pre><p>A specific version can also be loaded if required:</p><pre>$ module load samtools/1.4.1\n</pre><p>If no version is specified then a default is chosen as indicated in the available software list (usually the latest version). If there is a conflict you can remove the module from your environment:</p><pre>$ module load samtools/1.4.1\n</pre><p>If no version is specified then a default is chosen as indicated in the available software list (usually, but not always, the latest version). If there is a conflict you can remove the module from your environment:</p><pre>$ module unload samtools\n</pre><p>To unload all loaded modules:</p><pre>$ module purge\n</pre><p>You can see the modules you have loaded:</p><pre>$ module list\nCurrently Loaded Modulefiles:\n  1) samtools/1.5    2) varscan/2.3.9   3) R/3.4.0\n</pre><p>You can see how a particular module will modify the environment:</p><pre>$ module show anaconda3\n-------------------------------------------------------------------\n/stornext/System/data/modulefiles/bioinf/its/anaconda3/4.0.0:\n\nmodule-whatis\t Anaconda is a freemium distribution of the Python and R programming languages for large-scale data processing, predictive analytics, and scientific computing, that aims to simplify package management and deployment (v4.0.0) \nconflict\t anaconda3 \nsetenv\t\t PYTHONHOME /stornext/System/data/apps/anaconda3/anaconda3-4.0.0 \nprepend-path\t PATH /stornext/System/data/apps/anaconda3/anaconda3-4.0.0/bin \nprepend-path\t MANPATH /stornext/System/data/apps/anaconda3/anaconda3-4.0.0/share/man \nprepend-path\t CPATH /stornext/System/data/apps/anaconda3/anaconda3-4.0.0/include \nprepend-path\t LD_LIBRARY_PATH /stornext/System/data/apps/anaconda3/anaconda3-4.0.0/lib \nprepend-path\t LIBRARY_PATH /stornext/System/data/apps/anaconda3/anaconda3-4.0.0/lib \n-------------------------------------------------------------------\n</pre><p><br>Module configuration files can reside in a number of directories, for example, there are module directories for bioinformatics software and module directories for structural biology software. You may need to access modules in additional directories and you can instruct the module system to use those directories as follows:</p><pre>$ module use /stornext/System/data/modulefiles/bioinf/its\n</pre><p>When submitting batch jobs, your job script may need to issue some module commands so that the software required by the batch job is available. Module commands can also be included in the <span class="fontColorBlue"><em>.bashrc</em></span> so that your environment does not need to set up each time you log in.</p>
<h3><span class="fontColorThemeSecondary">Loading Modules in Programming Languages</span></h3>
