<h2>Moving data efficiently</h2><p>Datasets in the order of 10s TB are now fairly common and, of course, they will only get bigger. Network speeds, on the other hand, are around ~100MB/s, at best, and will unlikely change much in the foreseeable future. Downloading, uploading, or even moving data on-premises becomes a difficult task in and of itself that can take days or weeks. This guide provides some general advice for moving data. We also provide specific advice for downloading from the European Genome Archive (EGA).</p><p>Please contact <a data-cke-saved-href="mailto:research.computing@wehi.edu.au" href="mailto:research.computing@wehi.edu.au">Research Computing</a> if you need help or also contact us with any experience that may benefit other users.</p><h3><br></h3>
<h4><span class="fontColorBlue">Contents on this page:</span></h4><p><a href="/sites/rc2/SitePages/Moving-Data.aspx#moving-data-efficiently" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#moving-data-efficiently" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#moving-data-efficiently">Moving data efficiently</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#general-advice" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#general-advice" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#general-advice">General advice</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#downloading-from-the-european-genome-archive-%28ega%29" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#downloading-from-the-european-genome-archive-(ega)" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#downloading-from-the-european-genome-archive-(ega)">Downloading from the European genome archive</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#submitting-to-the-european-nucleotide-archive-%28ena%29" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#submitting-to-the-european-nucleotide-archive-(ena)" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#submitting-to-the-european-nucleotide-archive-(ena)">Submitting to the European nucleotide archive</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#create-an-account" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#create-an-account" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#create-an-account">Create an account</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#upload-data" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#upload-data" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#upload-data">Upload data</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#start-a-submission" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#start-a-submission" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#start-a-submission">Start a submission</a><br><a href="/sites/rc2/SitePages/Moving-Data.aspx#samples-page" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#samples-page" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Moving-Data.aspx#samples-page">Samples page</a></p>
<h3><span class="fontColorThemeSecondary">General advice</span></h3><ul><li>Use a protocol and software that will automatically resume an interrupted transfer, e.g. <a href="https://rsync.samba.org/" data-cke-saved-href="https://rsync.samba.org/">rsync</a>, <a href="https://aria2.github.io/" data-cke-saved-href="https://aria2.github.io/">aria2</a>, <a href="https://www.unix.com/man-page/redhat/1/lftp/" data-cke-saved-href="https://www.unix.com/man-page/redhat/1/lftp/" data-interception="on" title="https://www.unix.com/man-page/redhat/1/lftp/">lftp</a>, or if the other end supports it, <a href="http://download.asperasoft.com/download/docs/scp_client/2.5/aspera-client-unix.html" data-cke-saved-href="http://download.asperasoft.com/download/docs/scp_client/2.5/aspera-client-unix.html">Aspera</a>. All of these are available on WEHI systems (they may require a module).<br></li><li>For data movements that will take longer than a couple of hours, wrap them in a script and submit them to the batch system. Most transfers only require modest resources, e.g. 4 CPUs and 8GB of RAM.<br></li><li>Monitor the transfer to ensure that it has not stalled or failed for some reason.<br></li></ul><h3><span class="fontColorThemeSecondary">Downloading from the European Genome Archive (EGA)</span></h3><p>Recently, a researcher needed to download 6TB from EGA. EGA instructed them to use their proprietary client and protocol. Unfortunately, their software was badly flawed and caused the researcher significant delays and wasted time and effort. It turns out that EGA does in fact have an <a href="https://asperasoft.com/" data-cke-saved-href="https://asperasoft.com/">Aspera</a> server for their data and once the researcher had access the download proceeded smoothly.</p><p>If you are downloading from EGA please ask for an Aspera account. If EGA insists on you using their proprietary solution, please contact <a href="mailto:research.computing@wehi.edu.au" data-cke-saved-href="mailto:research.computing@wehi.edu.au">Research Computing</a> immediately. It is quite likely that they will fix their software but RC should be involved so we can be aware of current best practice.</p><h3><span class="fontColorThemeSecondary">Submitting to the European Nucleotide Archive (ENA)</span></h3><p>This procedure was developed by Jocelyn Penington from the Papenfuss Lab at WEHI.</p><h4><span class="fontColorBlue">Create an account:</span></h4><p><a href="https://www.ebi.ac.uk/ena/submit/sra/#registration" data-cke-saved-href="https://www.ebi.ac.uk/ena/submit/sra/#registration">https://www.ebi.ac.uk/ena/submit/sra/#registration</a>.</p><p>You will be allocated a username and asked for a password.</p><h4><span class="fontColorBlue">Upload data:</span></h4><ol><li><p>Make a directory of symbolic links so that all the data appears to be in one place. Not essential, but ensures all files have unique names without ‘/’.</p></li><li><p>All files need an md5 checksum generated. These can be supplied to ENA by uploading files with extension .md5 matching each filename of data, or by having md5 checksums as column(s) in the experiment spreadsheet.</p></li><li><p>All fastq and bam files (at least) need you to note insert size, instrument (e.g. Illumina MiSeq), and some description of the approach,<br>e.g. WGS, amplicon, etc.</p></li></ol><p>Data can be uploaded via the website, via ftp, or Aspera. Host to connect to is <span class="fontColorThemeSecondary"><strong>webin.ebi.ac.uk</strong></span>. For large amounts of data, Aspera was 20x faster than ftp. An example command is:</p><pre>ASPERA_SCP_PASS=&lt;your password&gt; ascp -L . \\\n-l10G \\\n--file-list=&lt;fl&gt; \\\n--policy=high \\\n--host=webin.ebi.ac.uk \\\n--mode=send \\\n--user=&lt;userid&gt; \\\n-d .</pre><p>where <em><span class="fontColorBlue">&lt;fl&gt;</span></em> is a file containing a list of fully qualified file names.</p><p>You will need to do this interactively once to accept the host key. Hit <span class="fontColorBlue">ctrl-C</span> and then submit as a batch job.</p>
<h4><span class="fontColorBlue">Start a submission:</span></h4><p>Partial submissions can’t be saved (although completed submissions can be easily added to) and submission can be an iterative process, so it is worth keeping copies of answers for easy re-entry.</p><p>The first page asks for a short name, a short descriptive title, and an abstract.</p><p>The next page asks for a date to make the submission public.</p>
<h4><span class="fontColorBlue">Samples page:</span></h4><ul><li>It is a good idea to populate the fields you want interactively for the first sample or two, then download the spreadsheet.</li><li>Make a note of which fields are required, and how any dropdown-list entries are translated into the tsv file.</li><li>Then (offline) edit the spreadsheet to list all samples for which you want the same descriptive columns. Delete any columns you don’t need but be sure to keep the required columns. Use the PREV button (not RESTART SUBMISSION) until you find the <em>Upload Spreadsheet</em> option and upload.</li><li>Check the sample descriptions, then proceed forward to <em>Runs</em>, and repeat.</li><li>Once again, it is a good idea to download a spreadsheet with the first row populated to check what kind of answers are required.</li><li>When you submit, the entries are checked against files uploaded to your area. If all checks pass, the submission is accepted and a study accession is generated.</li><li>Now you can add more samples and runs as needed.</li></ul><p><br><span class="fontColorRed"><span class="highlightColorGrey">New Submission &gt; choose the study &gt;&gt;Next</span></span><br>and proceed as above.</p>
