<h2>What is Nix?</h2><p>One of the bioinformaticians' most fundamental requirements is access to the software. Software drives our analyses, so software management is required for reproducibility. Unfortunately, software management is complex and error-prone, making it difficult to return to an analysis potentially years later and reproduce or extend.</p><p>The <strong>Nix</strong> package management system aims to solve issues encountered in software management and reproducibility through a unique approach with many <a data-interception="off" title="https://nixos.org/explore.html" href="https://nixos.org/explore.html" data-cke-saved-href="https://nixos.org/explore.html">advantages</a>. Nix enables reproducible software environments that are easily transferable to other machines. Multiple versions of software can coexist and interoperate, allowing the construction of complex pipelines that have specific runtime requirements (e.g., the Java Runtime Environment 8 for MuTect). It also has the <a href="https://repology.org/repositories/graphs" data-cke-saved-href="https://repology.org/repositories/graphs" data-interception="on" title="https://repology.org/repositories/graphs">most packages</a> available compared to other Linux package managers!</p><p>Must read: <a data-cke-saved-href="https://doi.org/10.1093/gigascience/giaa121" href="https://doi.org/10.1093/gigascience/giaa121">Article about BioNix Workflow</a></p><p>On Milton, Nix is supported by <a href="mailto:bedo.j@wehi.edu.au" data-cke-saved-href="mailto:bedo.j@wehi.edu.au" data-interception="on" title="mailto:bedo.j@wehi.edu.au">Justin Bedo</a> with help from the Research Computing Platform. If you need help, please contact the <a href="mailto:research.computing@wehi.edu.au" data-cke-saved-href="mailto:research.computing@wehi.edu.au" data-interception="on" title="mailto:research.computing@wehi.edu.au">Research Computing Platform</a>.</p><h3><span class="fontColorThemeSecondary">Getting Started with Nix on Milton</span></h3><p><strong>Before initiating the use of Nix on Milton, follow these steps</strong>:</p><ol><li>If you do not already have VAST scratch, please request it <strong><a href="https://support.wehi.edu.au/support/catalog/items/71" data-cke-saved-href="https://support.wehi.edu.au/support/catalog/items/71" target="_blank" data-interception="off" title="https://support.wehi.edu.au/support/catalog/items/71">here</a></strong>.</li><li>Submit a helpdesk ticket with the title "Requesting a nix-store directory on VAST".</li><li>Wait for confirmation from the team that your store has been created.</li><li>Read <a href="/sites/rc2/SitePages/Nix.aspx#managing-your-nix-store" data-cke-saved-href="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Nix.aspx#managing-your-nix-store" data-interception="on" title="https://wehieduau.sharepoint.com/sites/rc2/SitePages/Nix.aspx#managing-your-nix-store">this section</a> about managing and cleaning your nix-store</li></ol><p><strong>Please note</strong>, it's important that you do not start using the Nix module until the team has set up your store.&nbsp;</p><p>At WEHI, we now have a version of Nix available. To use it, load the module with the command</p><pre>module load nix/latest\n</pre><p>Which makes the nix command available to you.</p><p>On Milton, Nix is currently configured to use /vast/projects/nix/$USER as the store location. <strong>If you do not have a space in this directory, send an email requesting a nix store on VAST projects to support@wehi.edu.au.</strong></p><p>By default, Nix will use /vast/scratch/users/$USER/tmp to write temporary files. So, before running nix for the first time, make sure to create the temporary directory in your directory with the command:</p><pre>mkdir /vast/scratch/users/$USER/tmp</pre><p>which should be all the setup you need to do get nix running! Note that Nix on Milton will respect the TMPDIR environment variable if you wish to change where temporary files are written to.</p><p>When nix is run for the first time, it will create the directory</p><pre>/vast/projects/nix/$USER/nix/</pre><p>which will store all packages installed by Nix. You can delete this directory if you no longer wish to use Nix-installed packages.</p>
<p>Nix is primarily setup to be used on <strong>vc7-shared</strong> and will warn you in case you are not running on vc7-shared.</p>
<p>When using the nix module, you may experience error messages like:<br></p><pre><span></span>error: builder for '/nix/store/&lt;hash&gt;-&lt;package&gt;.drv' failed to produce output path for output 'out' at '&lt;hash&gt;-&lt;package&gt;'<span> </span></pre><p><span>This will be fixed in the next maintanence in late September 2023. Until then, if you encounter this error, you will need to restart the pipeline.</span><br></p>
<h3><span class="fontColorThemeSecondary">Working with your nix-store</span></h3><p>Your nix-store will be located in&nbsp;</p><pre>/vast/projects/nix/&lt;username&gt;</pre><p>It has a quota of 100GB.</p><p>Each user has access to their own personal Nix store. To setup your store to behave <em>like</em> you had Nix installed on your own computer, with the nix/latest module loaded, execute the command</p><pre>nix-chroot bash</pre><p>This will start a new bash shell where you will see that your store is located at /nix.</p><h4>Cleaning your Nix Store</h4><p>As your nix store on VAST projects is initially allocated a 100GB quota, you need to regularly clean up the store.</p><p>Read <strong><a title="https://nixos.wiki/wiki/Storage_optimization" data-interception="off" data-cke-saved-href="https://nixos.wiki/wiki/Storage_optimization" href="https://nixos.wiki/wiki/Storage_optimization">this article </a></strong>for more details on storage optimisation and also see the <a href="https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-store-gc.html" data-cke-saved-href="https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-store-gc.html" data-interception="on" title="https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-store-gc.html">garbage collection manual page</a>.</p>
<h3><span class="fontColorThemeSecondary">Running your first BioNix pipeline</span></h3><p>Try out your first BioNix pipeline by running the <a href="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe" data-cke-saved-href="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe" data-interception="on" title="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe">QC pipeline</a> ported from Snakemake:</p><pre>$ nix build github:WEHI-ResearchComputing/BioNix-qc-pipe<span>?</span>dir=qc-pipe/biobloom\n</pre><p>This will build the expressions from the biobloom subdirectory under <a href="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe" data-cke-saved-href="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe" title="https://github.com/WEHI-ResearchComputing/BioNix-qc-pipe" data-interception="on">the BioNix-qc-pipe repository</a>. This step may take a while to finish. When complete, the terminal will return quietly. The above command builds BioBloom tools in BioNix. you can then compare the cryptographic hash of result against the expected value to verify that the correct output has been generated:</p><pre>$ sha256sum -c &lt;(echo "817e25ba80dcbb89b8a5b5e9ba48dc82eba38f1cb54ae989fd18d0a6306b1718" ./result)</pre><p>The sha256sum -c will check if the hash being redirected to it matches the hash of the generated ./result file. A match indicates the file is identical to the expected output.</p><p>Note: if you didn't execute nix-chroot bash, the "result" link will be broken and the sha256sum check will fail!</p>
<h3>​​<span class="fontColorThemeSecondary">Nix and Slurm</span></h3><p>Nix on Milton has been patched to integrate with the Slurm scheduler. This means that Nix will submit jobs to the queue to build or execute software. This allows Nix to utilize more cores than what is usually available on your PC and consequently may speed up build and execution steps. Note that if the Slurm queue is busy, Nix may have to wait for resources to be allocated. For long running nix builds/runs, you may wish to use tools like tmux/screen to keep your session going in case you wish to disconnect from Milton.</p><p>By default, Slurm jobs submitted by Nix will request 2GB of memory. If you find your jobs are failing with an out-of-memory error, you can increase the requested memory with the environment variable:</p><pre>export NIX_SLURM_MEMORY=20G</pre><p>which will request 20GB of memory. You can modify the number as needed.</p><p>&lt;other features important environment variables?&gt;</p>
<h3><span class="fontColorThemeSecondary">Nix Basic Usage</span></h3><p>Like with many other shell commands, you can get help with using the command with</p><pre>nix --help</pre><p>You can also use</p><pre>nix &lt;nix-subcommand&gt; --help</pre><p>to get more information for a specific nix subcommand, where you substitute &lt;nix-subcommand&gt; with the subcommand of interest.</p><h4>Interactive shells with Nix packages</h4><p>Interactive shells can be created using the&nbsp;shell&nbsp;subcommand, for example:</p><pre>nix shell nixpkgs#bwa</pre><p>will drop the user into a shell containing BWA. Multiple packages can be specified, for example:</p><pre>nix shell nixpkgs#{bwa,samtools}</pre><p>Instead of dropping into a shell, a command can be specified. This is handy for use in scripts, or if the shell itself is unnecessary:</p><pre>[bedo.j@vc7-shared ~]$ nix shell nixpkgs#hello -c hello\nHello, world!\n[bedo.j@vc7-shared ~]$</pre><div><h4>Searching for software</h4></div><p>It is possible to search the nixpkgs repository for software that matches keywords using the&nbsp;nix search nixpkgs&nbsp;command. Here’s an example searching for the keyword&nbsp;bowtie:</p><pre>[bedo.j@vc7-shared ~]$ nix search nixpkgs bowtie\n* legacyPackages.x86_64-linux.bowtie (1.3.1)\nAn ultrafast memory-efficient short read aligner\n\n* legacyPackages.x86_64-linux.bowtie2 (2.4.4)\nAn ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences\n\n* legacyPackages.x86_64-linux.python38Packages.genome-collector (0.1.6)\nGenomes and build BLAST/Bowtie indexes in Python\n\n* legacyPackages.x86_64-linux.python39Packages.genome-collector (0.1.6)\nGenomes and build BLAST/Bowtie indexes in Python\n[bedo.j@vc7-shared ~]$\n</pre><p>Two versions of bowtie (1.3.1 and 2.4.4) are available under the names bowtie and bowtie2, respectively, as is a python package that operates on bowtie indices.</p><h4>nix run</h4><p><span class="fontColorRed"><strong>Coming SOON!</strong></span></p><h4>nix build</h4><p><span class="fontColorRed"><strong>Coming SOON!</strong></span></p>
